util/docker.js

```diff
@@ -0,0 +1,187 @@
+import { confirm, input } from "@inquirer/prompts";
+
+function validatePortFormat(input) {
+  const validFormat = /^\d+:\d+$/;
+  return validFormat.test(input) || "Ports must be in the format hostPort:containerPort.";
+}
+
+async function promptForEnvironmentVariables() {
+  const addEnvironmentVariables = await confirm({
+    message: "Do you want to add environment variables?",
+    default: false,
+  });
+
+  if (addEnvironmentVariables) {
+    const variableInput = await input({
+      message: "Enter environment variables separated by commas (e.g., VAR1=DB_USER, VAR2=DB_PASS):",
+      validate: (input) => {
+        // Regex Expression to Evaluate User Input in Format: VAR1=DB_USER, VAR2=DB_PASS.
+        const validFormat = /^([a-zA-Z_][a-zA-Z0-9_]*=[a-zA-Z_][a-zA-Z0-9_]*)(,\s*[a-zA-Z_][a-zA-Z0-9_]*=[a-zA-Z_][a-zA-Z0-9_]*)*$/;
+        return (
+          validFormat.test(input) ||
+          "Input must be in the format VAR1=DB_USER, VAR2=DB_PASS (comma-separated pairs)."
+        );
+      },
+    });
+
+    const environmentVariables = {};
+    variableInput.split(",").forEach((pair) => {
+      const [key, envVariable] = pair.split("=").map((item) => item.trim());
+      environmentVariables[key] = envVariable;
+    });
+
+    return environmentVariables;
+  }
+
+  return undefined;
+}
+
+async function promptAppService(packageName) {
+  const image = await input({
+    message: "Enter the Docker image for the app service (default: node:20-alpine):",
+    default: "node:20-alpine",
+  });
+
+  const ports = await input({
+    message: "Enter the ports to expose for the app service (default: 8080:8080):",
+    default: "8080:8080",
+    validate: validatePortFormat,
+  });
+
+  const environment = await promptForEnvironmentVariables();
+
+  return {
+    name: packageName,
+    image,
+    containerName: `${packageName}_container`,
+    ports: [ports],
+    environment,
+  };
+}
+
+async function promptDbService(packageName) {
+  const image = await input({
+    message: "Enter the Docker image for the database (default: mysql:latest):",
+    default: "mysql:latest",
+  });
+
+  const ports = await input({
+    message: "Enter the ports to expose for the database service (default: 5432:5432):",
+    default: "5432:5432",
+    validate: validatePortFormat,
+  });
+
+  const environment = await promptForEnvironmentVariables();
+
+  return {
+    name: `${packageName}_db`,
+    image,
+    containerName: `${packageName}_db_container`,
+    ports: [ports],
+    environment,
+  };
+}
+
+async function promptCacheService(packageName) {
+  const image = await input({
+    message: "Enter the Docker image for the cache service (default: redis:latest):",
+    default: "redis:latest",
+  });
+
+  const ports = await input({
+    message: "Enter the ports to expose for the cache service (default: 6379:6379):",
+    default: "6379:6379",
+    validate: validatePortFormat,
+  });
```

Comment:
need not be specified through input, let it use default as whatever we're giving is a starter template. I don't think in that this level of customization is needed.